tools/aa/aa_tool.sh

```diff
@@ -61,13 +66,60 @@ function find_base_folder {
   return 0
 }
 
+function local_server {
+    # starts local_server if it is not running yet
```

Comment:
Can you describe on a high level how you're doing it?
and then mention shortly before each step.